File: app-flutter/lib/router/imageRouter.dart
import 'package:flutter_web/material.dart';
// ======================================================================================================== 路由
// ======================================================================================================== Image widget
import 'package:flutter_web_study.app/pages/flutter/widget/image/widget/Image.dart';
import 'package:flutter_web_study.app/pages/flutter/widget/image/widget/ImageAsset.dart';
import 'package:flutter_web_study.app/pages/flutter/widget/image/widget/ImageNetwork.dart';
import 'package:flutter_web_study.app/pages/flutter/widget/image/widget/FadeInImage.dart';
import 'package:flutter_web_study.app/pages/flutter/widget/image/widget/FadeInImage_assetNetwork.dart';
// ======================================================================================================== x
import 'package:flutter_web_study.app/pages/flutter/widget/image/style/x/BoxFit.dart';
import 'package:flutter_web_study.app/pages/flutter/widget/image/style/x/ImageRepeat.dart';
import 'package:flutter_web_study.app/pages/flutter/widget/image/style/x/BlendMode.dart';
import 'package:flutter_web_study.app/pages/flutter/widget/image/style/x/ColorFilter.dart';
import 'package:flutter_web_study.app/pages/flutter/widget/image/style/x/FilterQuality.dart';
// ======================================================================================================== 路由类
class ImageRouter{
	final Map<String,WidgetBuilder> router = <String,WidgetBuilder>{
		// ======================================================================================================== Image
		'/ImagePage': (BuildContext context) => ImagePage(title:'Image( )'),
		'/ImageAssetPage': (BuildContext context) => ImageAssetPage(title:'Image.asset( )'),
		'/ImageNetworkPage': (BuildContext context) => ImageNetworkPage(title:'Image.network( )'),
		'/FadeInImagePage': (BuildContext context) => FadeInImagePage(title:'FadeInImage( )'),
		'/FadeInImageAssetNetworkPage': (BuildContext context) => FadeInImageAssetNetworkPage(title:'FadeInImage.assetNetwork( )'),

		// ======================================================================================================== x
		'/BoxFitPage': (BuildContext context) => BoxFitPage(title:'BoxFit.x'),
		'/ImageRepeatPage': (BuildContext context) => ImageRepeatPage(title:'ImageRepeat.x'),
		'/BlendModePage': (BuildContext context) => BlendModePage(title:'BlendMode.x'),
		'/ColorFilterPage': (BuildContext context) => ColorFilterPage(title:'ColorFilter.x'),
		'/FilterQualityPage': (BuildContext context) => FilterQualityPage(title:'FilterQuality.x'),
	};
}

File: app-react/src/pages/flutter/routers/imageRouter.js
/* ====================================== 模块子路由配置  ====================================== */
import React from 'react'
import { Switch, Route } from 'react-router-dom'
// ===================================================================== 异步加载
import Import from '@common/bundle'
// ===================================================================== 二级路由
export default class ImageRouter extends React.Component{
	render(){
		const { match } = this.props
		return (
				<Switch>
					{/* Image widget */}
					<Route path={`${match.url}/Image`} 						component={ Import('flutter/widget/image/widget/Image/Image') } exact />
					<Route path={`${match.url}/ImageAsset`} 				component={ Import('flutter/widget/image/widget/Image/Image.asset') } exact />
					<Route path={`${match.url}/ImageNetwork`} 				component={ Import('flutter/widget/image/widget/Image/Image.network') } exact />
					<Route path={`${match.url}/ImageFile`} 					component={ Import('flutter/widget/image/widget/Image/Image.file') } exact />
					<Route path={`${match.url}/ImageMemory`} 				component={ Import('flutter/widget/image/widget/Image/Image.memory') } exact />
					{/* FadeInImage widget */}
					<Route path={`${match.url}/FadeInImage`} 				component={ Import('flutter/widget/image/widget/FadeInImage/FadeInImage') } exact />
					<Route path={`${match.url}/FadeInImageAssetNetwork`} 	component={ Import('flutter/widget/image/widget/FadeInImage/FadeInImageAssetNetwork') } exact />
					<Route path={`${match.url}/FadeInImageMemoryNetwork`} 	component={ Import('flutter/widget/image/widget/FadeInImage/FadeInImageMemoryNetwork') } exact />
					{/* style */}
					<Route path={`${match.url}/DecorationImage`} 			component={ Import('flutter/widget/image/style/DecorationImage') } exact />
					<Route path={`${match.url}/AssetImage`} 				component={ Import('flutter/widget/image/style/AssetImage') } exact />
					<Route path={`${match.url}/NetworkImage`} 				component={ Import('flutter/widget/image/style/NetworkImage') } exact />
					<Route path={`${match.url}/BoxFit`} 					component={ Import('flutter/widget/image/style/x/BoxFit') } exact />
					<Route path={`${match.url}/ImageRepeat`} 				component={ Import('flutter/widget/image/style/x/ImageRepeat') } exact />
					<Route path={`${match.url}/BlendMode`} 					component={ Import('flutter/widget/image/style/x/BlendMode') } exact />
					<Route path={`${match.url}/ColorFilter`} 				component={ Import('flutter/widget/image/style/x/ColorFilter') } exact />
					<Route path={`${match.url}/FilterQuality`} 				component={ Import('flutter/widget/image/style/x/FilterQuality') } exact />
				</Switch>
		)
	}
}